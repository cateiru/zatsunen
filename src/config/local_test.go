package config_test

import (
	"testing"

	"github.com/cateiru/zatsunen/src/config"
	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/require"
)

func TestGetMode(t *testing.T) {
	local := config.SetLocalConfig("/path")

	require.Equal(t, local.GetMode(), "local")
}

func TestGetConfig(t *testing.T) {
	local := config.SetLocalConfig("/path")

	snaps.MatchSnapshot(t, local.GetConfig())
}

func TestSetLocalConfig(t *testing.T) {
	local := config.SetLocalConfig("/path")

	snaps.MatchSnapshot(t, local)
}
