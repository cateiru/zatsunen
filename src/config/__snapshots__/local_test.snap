
[TestGetConfig - 1]
config.Config{
    Mode:         "local",
    CommonConfig: config.CommonConfig{},
    LogConfig:    config.LogConfig{
        Options: &slog.HandlerOptions{
            AddSource:   true,
            Level:       slog.Level(-4),
            ReplaceAttr: func([]string, slog.Attr) slog.Attr {...},
        },
        IsJsonLog: false,
    },
    MiddlewareConfig: config.MiddlewareConfig{
        AllowSecFetchSiteValues: {},
    },
}
---

[TestSetLocalConfig - 1]
config.Local{
    Config: config.Config{
        Mode:         "local",
        CommonConfig: config.CommonConfig{},
        LogConfig:    config.LogConfig{
            Options: &slog.HandlerOptions{
                AddSource:   true,
                Level:       slog.Level(-4),
                ReplaceAttr: func([]string, slog.Attr) slog.Attr {...},
            },
            IsJsonLog: false,
        },
        MiddlewareConfig: config.MiddlewareConfig{
            AllowSecFetchSiteValues: {},
        },
    },
}
---
